test/Microsoft.AspNet.Mvc.ModelBinding.Test/Binders/KeyValuePairModelBinderTest.cs

```diff
@@ -28,8 +26,7 @@ public class KeyValuePairModelBinderTest
             var retVal = await binder.BindModelAsync(bindingContext);
 
             // Assert
-            Assert.NotNull(retVal);
-            Assert.Null(retVal.Model);
+            Assert.Null(retVal);
```

Comment:
I made a few changes (2 or 3 commits) as I went through trying to understand the system.  this was part of that -- the multiple locations where `isModelSet: false` was used but just didn't matter was getting in the way and made the system less understandable.

if you don't like it, I can revert it.  as I said [elsewhere](https://github.com/aspnet/Mvc/pull/2188#discussion_r26820011), that'll take some additional time (hour or so) and I'm done tonight almost.